### PR TITLE
Change icons in web UI

### LINKED ISSUE #1469: Consider: SVG for icons?
Really enjoying Mastodon's excellent user interface. I've noticed the interface uses icons in a critical manor, so I think browser support here is important.

## Support Analytics
SVG actually has a better caniuse score than webfonts. 

<details>
  <summary>SVG 97.96% safe (screenshot)</summary>
  
</details>
<br>

<details>
  <summary>Webfonts 95.3% safe (screenshot)</summary>
  
</details>
<br>

To be fair, while SVG itself is over 97% safe to use the SVG `<use>` syntax is not as safe:
> IE9-Edge12, Safari 5.1-6, and UCWeb 11 do not support [referencing external files](https://css-tricks.com/svg-use-external-source/) via <use xlink:href>. Polyfills are available: [server-side inlining + snippet](http://codepen.io/hexalys/pen/epErZj/) - [script](https://github.com/jonathantneal/svg4everybody)

But polyfills take SVG `<use>` back up to 97.96% safe. 

_Note: Font Awesome 5 will ship with an SVG framework_

## The Un&ndash;Detectables 
Users may leverage the user agent with user sheets or plugins that manipulate a web app's style to be more legible to them. Here is what the interface looks like when the FontAwesome font-face is overridden:
<details>
  <summary>UI w/ icon font overriden (screenshot)</summary>
  Mastodon Interface with empty squares where Icons should be
</details>
<br>

We have no way of detecting if a user has done this, so truth be told, web fonts are undetectably less than 95% safe to use.

## Technical Difference

We'd be changing this:
```html
<i class="fa fa-fw fa-globe"></i>
```

to something like this:
```html
<svg class="icon icon-globe">
  <title>Federated Timeline</title>
  <use xlink:href="assets/img/icons.svg#icon-globe"></use>
</svg>
```

## Pros
 - better browser support
   - possible to [achieve 100% browser support with an `<image>` fallback](https://css-tricks.com/a-complete-guide-to-svg-fallbacks/)
 - better Art Direction
    - the individual shapes of icons, not just the entire icons themselves, can be styled and animated
    - the SVG sprite sheet can be generated from a folder of SVG source files, so it is easy to edit them or add custom icons
 - optimization, easier to include just the icons you need
 - `<SVG>` is more semantic for an icon. `<SVG>` is for graphics, `<i>` is for oblique text

## Cons
 - icon sprite sheet can't be served from another domain (doesn't seem to matter in our case)
 - need to use a JS polfyill for IE (Mastodon requires JS anyways)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### LINKED ISSUE #26617: Change Icon Library
### Pitch

Hi, 
I am using the Remix Icon Library, which is more diverse and better looking than Font Awesome. Can Mastodon use this icon library?

截屏2023-08-23 08 06 49

### Motivation

It is also more convenient for users to customize icons.

### PR DESCRIPTION
**Pull request by Gargron**
We're finally replacing Font Awesome with Material icons, consistent with our Android app. For now this is a pretty straightforward mapping of old icons to new icons, but Material gives us more possibilities (e.g. filled in for active, outline for inactive) that I imagine we'll explore gradually afterwards.

image

___

Fixes MAS-166, fixes #26617, fixes #1469

### CODE REVIEW

**Review by ykzts — DISMISSED** [review]

**Comment by vmstan** [comment]
Gave this PR a test run, and the icon alignment in post views is way off in Safari.

CleanShot 2023-10-18 at 16 27 39@2x
CleanShot 2023-10-18 at 16 26 02@2x
CleanShot 2023-10-18 at 16 25 22@2x

**Comment by vmstan** [comment]
image

More problematic on mobile.

**Comment by vmstan** [comment]
Safari issue is fixed.

**Comment by vmstan** [comment]
image

Reply content with client info appears misaligned for longer client names.

**Comment by vmstan** [comment]
Client informed appears fixed now.

**Comment by ClearlyClaire** [comment]
The dropdown menu icon is definitely misaligned. Also, all icons on posts seem slightly too small. Finally, animations (fav icon and boost icon) are missing. The outline VS filled distinction is welcome, but it's not there for boosts, which has also lost its variants and animations.

I'm also running into an infinite loop in the advanced interface.

File: app/javascript/mastodon/features/ui/components/column_loading.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Column from 'mastodon/components/column';
+import ColumnHeader from 'mastodon/components/column_header';
+
+export const ColumnLoading: React.FC = () => (
+  <Column>
+    <ColumnHeader />
+    <div className='scrollable' />
+  </Column>
+);
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/ui/components/column_loading.tsx:9]: This incorrectly drops the `multiColumn` prop, sometimes causing an infinite loop in advanced mode.

**Review by ClearlyClaire — DISMISSED** [review]

**Comment by vmstan** [comment]
> I'm also running into an infinite loop in the advanced interface.

Was just coming here to say this.

File: app/javascript/mastodon/features/ui/components/column_loading.tsx (hunk 1 above)
Comment (Gargron) [reply app/javascript/mastodon/features/ui/components/column_loading.tsx:9]: I did it because the type check was throwing errors about it. Hm.

**Comment by Gargron** [comment]
> Also, all icons on posts seem slightly too small.

Material Symbols include padding in the SVGs which means the actual shapes are smaller than the equivalent Font Awesome symbols at the same container size. However, I have adjusted the sizes to be closer to what they used to be. I believe any remaining impression of smallness is just from them being a bit thinner design-wise.

> Finally, animations (fav icon and boost icon) are missing

I won't re-add them. Favourite icon is about to become a heart, so spinning will make no sense anymore. There's a ticket for Sam to create interaction animations for the iOS/Android apps, once that exists I imagine we'll be able to port it back to web.

**Comment by vmstan** [comment]
I like the new weights.

**Review by renchap — APPROVED** [review]
LGTM after testing on my own instance

**Comment by colinstu12** [comment]
> I won't re-add them. Favourite icon is about to become a heart, so spinning will make no sense anymore. There's a ticket for Sam to create interaction animations for the iOS/Android apps, once that exists I imagine we'll be able to port it back to web.

Where was it discussed or decided that the star should become a heart? The star is a lot more generic of an interaction than a heart is (that indicates liking/loving something). Not exactly the nicest thing to use with posts that contain bad or sad news/topics, etc. 
Twitter made this same move too long ago and didn’t seem appropriate either.

**Comment by edoswald** [comment]
I understand we're changing the star to a heart, but in a thread on this on Mastodon came up with some interesting real-world data on the "star." It seems that people are using it for a variety of uses, some duplicating pre-existing features, like bookmarks. Changing it to a heart will also likely change its usage since the heart is used for status reactions on most other networks, so newbies will likely do it here.

Maybe before changing the star to a heart, maybe taking a look at the functionality the star provides, and rethinking it? I know this is a simple change, but it may result in some unintended consequences from an interaction standpoint, some of which may not be what the devs intended.

I think the base problem is nobody knows exactly what the star is for, or does. I'd be strongly in favor of considering a change to how the star works before such a switch.

**Comment by edoswald** [comment]
> > I won't re-add them. Favourite icon is about to become a heart, so spinning will make no sense anymore. There's a ticket for Sam to create interaction animations for the iOS/Android apps, once that exists I imagine we'll be able to port it back to web.
> 
> @Gargron Where was it discussed or decided that the star should become a heart? The star is a lot more generic of an interaction than a heart is (that indicates liking/loving something). Not exactly the nicest thing to use with posts that contain bad or sad news/topics, etc. Twitter made this same move too long ago and didn’t seem appropriate either.

I think it's fine. Meta decided to overcomplicate it with four other options. I also think the average consumer is mature enough to understand that the like is for the content, not necessarily the topic.

**Comment by brendanjones** [comment]
Please don’t change the star to a heart. Every time I go to favourite a post about a negative topic I’ll feel resistance. It won’t be the end of the world, but I won’t like it.

**Comment by ronilaukkarinen** [comment]
I don't know if this is the place to chime in about the stars vs. hearts, but I think it's fine. I really use the the star like a heart already (1. Acknowledged, I have read your post, 2. Sending love/understanding/agreement).

**Comment by edoswald** [comment]
I think you’re in the right place.

**Comment by Squaredude** [comment]
Adding my two cents on this matter.

My biggest concern with the usage of a "heart" icon is that it predisposes a certain mentality/emotion. As others have said, one can feel awkward "liking" / "hearting" ("loving"?) a post with tragic content.

But, on the abstract side of things, it seems manipulative, like something a for-profit would do.

One thing I've been liking about Mastodon is that the interface and the environment of the Fediverse is not algorithmic, not commercial, and therefore there is no need to encourage the users into having emotional interactions. "Bookmarks", "boosts", "stars"/"favorites", "posts"... these are neutral terms, they don't invoke emotions or predispose users to interact.

I hope the measure can be reconsidered. It's frankly disappointing that this was decided.

**Comment by brendanjones** [comment]
I realise this is merged already, but wanted to ask why the Explore menu item doesn't use the Material explore icon? 

1. It is the 'correct' icon for such a menu item (at least on Android). 
2. The Explore menu doesn't only let you explore hashtags, so using a hashtag icon excludes representing exploring posts, people and news.
3. Not using the hashtag icon would free up that icon to be used for https://github.com/mastodon/mastodon/issues/11091

I know Twitter used the hashtag icon for explore so people are familiar with it, but we don't need to always follow what Twitter does, do we? :)

**Pull request merged**